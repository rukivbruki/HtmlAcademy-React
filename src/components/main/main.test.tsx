import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {BrowserRouter as Router} from 'react-router-dom';
import NameSpace from "../../reducer/name-space.js";
import Main from './main';

const mockStore = configureStore([]);

const promoData = {
  name: `promoName`,
  genre: `promoGenre`,
  releaseDate: 0,
};

const films = [
  {
    id: 1,
    name: `name1`,
    img: `img/name1.jpg`,
  },
  {
    id: 2,
    name: `name2`,
    img: `img/name2.jpg`,
  },
  {
    id: 3,
    name: `name3`,
    img: `img/name3.jpg`,
  },
];

const user = {
  id: 1,
  email: `test@email.com`,
  name: `test user`,
  avatarUrl: `avatar`
};

it(`Main renders correctly`, () => {
  const store = mockStore({
    [NameSpace.GENRES]: {
      genres: [`genre1`, `genre2`],
      genreFilter: `All genres`
    },
    [NameSpace.FILMS]: {
      films
    },
    [NameSpace.USER]: {
      user
    },
    [NameSpace.FAVORITES]: {
      favorites: films
    }
  });

  const tree = renderer
    .create(
        <Router>
          <Provider store={store}>
            <Main promoData={promoData} filteredFilms={films}/>
          </Provider>
        </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
