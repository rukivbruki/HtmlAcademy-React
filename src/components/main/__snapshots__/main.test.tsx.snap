// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="promoName"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="sign-in__link"
          href="/login"
          onClick={[Function]}
        >
          Sign In
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            promoName
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              promoGenre
            </span>
            <span
              className="movie-card__year"
            />
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                fill="#d9cd8d"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 5v14l11-7z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            genre1
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            genre2
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="name1"
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "display": "flex",
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              name1
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="name2"
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "display": "flex",
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              name2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="name3"
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "flex-end",
                  "display": "flex",
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              name3
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
